http_handlers.go

```diff
@@ -102,6 +141,109 @@ func getMetadata(filter string) http.HandlerFunc {
 	}
 }
 
+func ec2Handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	logger.Debug("Calling ec2Handler ")
+	userIP := getIPFromRequest(r)
+	if userIP == "" {
+		return
+	}
+
+	metrics.MetadataRequests.Inc()
+	logger.With("userIP", userIP).Info("Actual IP is: ")
+	hw, err := getByIP(context.Background(), hegelServer, userIP) // returns hardware data as []byte
+	if err != nil {
+		metrics.Errors.WithLabelValues("metadata", "lookup").Inc()
+		logger.Info("Error in finding hardware: ", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	res, err := processEC2Query(r.URL.Path)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("404 not found"))
+		if err != nil {
+			logger.Error(err, "failed to write error response")
+		}
+	}
+
+	var resp []byte
+	if filter, ok := res.(string); ok {
+		resp, err = filterMetadata(hw, filter)
+		if err != nil {
+			logger.Error(err, "Error in filtering metadata: ")
+		}
+	} else if submenu, ok := res.(map[string]interface{}); ok {
+		var keys []string
+		for item := range submenu {
+			switch item {
+			case "_base": // _base is only used to keep track of the base filter, not a metadata item
+				continue
+			case "spot": // list only if instance is spot
+				resp, err := filterMetadata(hw, ".metadata.instance.spot")
```

Comment:
Is this okay to be hardcoded? Should it refer to the `ec2Filters` map instead? 

**Pros:** changes to the `spot` filter inside `ec2Filters` will automatically change the filter here as well  
**Cons:** To build that same path, I would need to do `submenu["_base"] + submenu[item]["_base"]`, which is harder to read.